File: example-kotlin-java/src/main/java/com/useek/example_kotlin_java/MainActivity.java
package com.useek.example_kotlin_java;

import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import com.useek.library_kt_beta.USeekManager;
import com.useek.library_kt_beta.USeekPlayerActivity;

import kotlin.Unit;
import kotlin.jvm.functions.Function3;

public class MainActivity extends AppCompatActivity {

    Button buttonRequestPoints;
    
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        USeekManager.Companion.getSharedInstance().setPublisherId("60d95e35d89800b0ee499e60d0735fb8");

        setContentView(R.layout.activity_main);

        Button button = findViewById(R.id.buttonActivity);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onPressPlayActivity();
            }
        });

        Button button1 = findViewById(R.id.buttonFragment);
        button1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onPressedFragmentSampleActivity();
            }
        });

        Button button2 = findViewById(R.id.buttonCustomView);
        button2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onPressedCustomViewSampleActivity();
            }
        });

        Button button3 = findViewById(R.id.buttonProgrammaticallyCustomView);
        button3.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onPressedProgrammaticallyCustomViewSampleActivity();
            }
        });

        buttonRequestPoints = findViewById(R.id.buttonRequestPoints);
        buttonRequestPoints.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onPressedRequestPoint();
            }
        });
    }

    public void onPressPlayActivity() {

        ExampleSettingsManager settingsManager = ExampleSettingsManager.sharedInstance();

        /*
        USeekPlayerActivity.setUSeekPlayerCloseListener(new USeekPlayerCloseListener() {
            @Override
            public void useekPlayerDidClosed(USeekPlayerView useekPlayerView) {
                Log.d("USeek Sample", "didClose()");
                promptRequestPoints();
            }

            @Override
            public void useekPlayerDidFailWithError(USeekPlayerView useekPlayerView, WebResourceError error) {
                Log.d("USeek Sample", "useekPlayerDidFailWithError()");
            }

            @Override
            public void useekPlayerDidStartLoad(USeekPlayerView useekPlayerView) {
                Log.d("USeek Sample", "useekPlayerDidStartLoad()");
            }

            @Override
            public void useekPlayerDidFinishLoad(USeekPlayerView useekPlayerView) {
                Log.d("USeek Sample", "useekPlayerDidFinishLoad()");
            }
        });
        USeekPlayerActivity.setCloseButtonHidden(!settingsManager.isShowCloseButton());
        USeekPlayerActivity.setLoadingText(settingsManager.getLoadingText());
        */

        Intent intent = new Intent(this, USeekPlayerActivity.class);
        intent.putExtra(USeekPlayerActivity.Companion.getUSEEK_USER_ID(), settingsManager.getUserId());
        intent.putExtra(USeekPlayerActivity.Companion.getUSEEK_GAME_ID(), settingsManager.getGameId());
        startActivity(intent);
    }

    public void onPressedFragmentSampleActivity() {
        Intent intent = new Intent(this, FragmentSampleActivity.class);
        startActivity(intent);
    }

    public void onPressedCustomViewSampleActivity() {
        Intent intent = new Intent(this, CustomViewSampleActivity.class);
        startActivity(intent);
    }

    public void onPressedProgrammaticallyCustomViewSampleActivity() {
        Intent intent = new Intent(this, CustomViewProgrammaticallyActivity.class);
        startActivity(intent);
    }

    public void onPressedRequestPoint() {
        ExampleSettingsManager settingsManager = ExampleSettingsManager.sharedInstance();
        buttonRequestPoints.setText("Please wait...");
        buttonRequestPoints.setEnabled(false);
        Function3<? super Integer, ? super Integer, ? super Error, Unit> requestPointLambda =
                new Function3<Integer, Integer, Error, Unit>() {
                    @Override
                    public Unit invoke(Integer lastPlayPoints, Integer totalPlayPoints, Error error) {
                        if (error == null) {
                            Toast.makeText(
                                    MainActivity.this,
                                    String.format("Your last play points : %d\nYour total play points : %d", lastPlayPoints, totalPlayPoints),
                                    Toast.LENGTH_LONG)
                                    .show();
                        } else {
                            Toast.makeText(
                                    MainActivity.this,
                                    error.getLocalizedMessage(),
                                    Toast.LENGTH_LONG)
                                    .show();
                        }
                        buttonRequestPoints.setText("Request Latest Points");
                        buttonRequestPoints.setEnabled(true);
                        return null;
                    }
                };
        USeekManager.Companion.getSharedInstance().requestPoints(
                settingsManager.getGameId(),
                settingsManager.getUserId(),
                requestPointLambda
        );

    }

    public void onPressedSettings() {
        Intent intent = new Intent(this, SettingsActivity.class);
        startActivity(intent);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_main_tab, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            onPressedSettings();
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

    private void promptRequestPoints() {
        AlertDialog alertDialog = new AlertDialog.Builder(MainActivity.this).create();
        alertDialog.setTitle("Confirmation");
        alertDialog.setMessage("Will you request for your latest points?");
        alertDialog.setButton(AlertDialog.BUTTON_POSITIVE, "YES",
                new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int which) {
                        dialog.dismiss();
                        onPressedRequestPoint();
                    }
                });
        alertDialog.setButton(DialogInterface.BUTTON_NEGATIVE, "NO", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                dialog.dismiss();
            }
        });
        alertDialog.show();
    }
}

File: example-kotlin/src/main/java/com/useek/example_kotlin/kotlin_kotlin/MainActivity.kt
package com.useek.example_kotlin.kotlin_kotlin

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.Toast
import com.useek.example_kotlin.ExampleSettingsManager
import com.useek.example_kotlin.R
import com.useek.example_kotlin.SettingsActivity
import com.useek.library_kt_beta.USeekManager
import com.useek.library_kt_beta.USeekPlayerActivity
import com.useek.library_kt_beta.USeekPlayerActivity.Companion.USEEK_GAME_ID
import com.useek.library_kt_beta.USeekPlayerActivity.Companion.USEEK_USER_ID
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        USeekManager.sharedInstance.publisherId = "60d95e35d89800b0ee499e60d0735fb8"

        buttonActivity.setOnClickListener { onPressPlayActivity() }

        buttonFragment.setOnClickListener { onPressedFragmentSampleActivity() }

        buttonCustomView.setOnClickListener { onPressedCustomViewSampleActivity() }

        buttonProgrammaticallyCustomView.setOnClickListener { onPressedProgrammaticallyCustomViewSampleActivity() }

        buttonRequestPoints.setOnClickListener { onPressedRequestPoint() }
    }

    private fun onPressPlayActivity() {

        val settingsManager = ExampleSettingsManager.sharedInstance

        val intent = Intent(this, USeekPlayerActivity::class.java)
        intent.putExtra(USEEK_USER_ID, settingsManager.userId)
        intent.putExtra(USEEK_GAME_ID, settingsManager.gameId)
        startActivity(intent)
    }

    private fun onPressedFragmentSampleActivity() {
        val intent = Intent(this, FragmentSampleActivity::class.java)
        startActivity(intent)
    }

    private fun onPressedCustomViewSampleActivity() {
        val intent = Intent(this, CustomViewSampleActivity::class.java)
        startActivity(intent)
    }

    private fun onPressedProgrammaticallyCustomViewSampleActivity() {
        val intent = Intent(this, CustomViewProgrammaticallyActivity::class.java)
        startActivity(intent)
    }

    private fun onPressedRequestPoint() {
        val settingsManager = ExampleSettingsManager.sharedInstance
        if (settingsManager.gameId == null) return
        buttonRequestPoints.text = "Please wait..."
        buttonRequestPoints.isEnabled = false
        USeekManager
                .sharedInstance
                .requestPoints(
                        settingsManager.gameId!!,
                        settingsManager.userId
                ) { lastPlayPoints, totalPlayPoints, error ->

                    if (error == null) {
                        Toast.makeText(
                                this@MainActivity,
                                String.format("Your last play points : %d\nYour total play points : %d", lastPlayPoints, totalPlayPoints),
                                Toast.LENGTH_LONG
                        ).show()
                    } else {
                        Toast.makeText(
                                this@MainActivity,
                                error.toString(),
                                Toast.LENGTH_LONG
                        ).show()
                    }
                    buttonRequestPoints.text = "Request Latest Points"
                    buttonRequestPoints.isEnabled = true

                }
    }

    private fun onPressedSettings() {
        val intent = Intent(this, SettingsActivity::class.java)
        startActivity(intent)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_main_tab, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val id = item.itemId

        if (id == R.id.action_settings) {
            onPressedSettings()
            return true
        }

        return super.onOptionsItemSelected(item)
    }

}
